### PR TITLE
Initial implementation to make basic specs pass

### PR DESCRIPTION
**Pull request by SwathiSankararaman**

### CODE REVIEW

File: lib/ucfg.rb [hunk 1]
```diff
@@ -1,11 +1,49 @@
...
 
-module Ucfg
+module Ucfg # rubocop:todo Style/Documentation
   class Error < StandardError; end
-  # Your code goes here...
+  # rubocop:todo Metrics/PerceivedComplexity
+  # rubocop:todo Metrics/MethodLength
+  # rubocop:todo Metrics/AbcSize
+  def self.validate(config, schema) # rubocop:todo Metrics/CyclomaticComplexity
+    OpenStruct.new(valid?: true, errors: [])
```
Comment (orhantoy) [inline lib/ucfg.rb]: We can remove this line as we don't assign the object to a variable, meaning we basically are not using it.

File: lib/ucfg.rb [hunk 2]
```diff
@@ -1,11 +1,49 @@
...
+      end
+    end
+    # fails if string property is provided as other type
+    config.merge(schema["properties"]) do |key, oldvalue, newvalue|
+      if oldvalue.class != newvalue["type"].class
+        valid = false
+        errors << "Property `#{key}` must be of type `#{newvalue['type']}`(provided value `#{oldvalue}` of type `boolean`)"
+      end
+    end
+    OpenStruct.new(valid?: valid, errors: errors)
```
Comment (orhantoy) [inline lib/ucfg.rb:47]: Let's add an empty line before this one, just makes the it more obvious that this line here is responsible for the return value.

File: lib/ucfg.rb [hunk 3]
```diff
@@ -1,11 +1,49 @@
...
+        end
+      end
+    end
+    # fails if string property is provided as other type
+    config.merge(schema["properties"]) do |key, oldvalue, newvalue|
+      if oldvalue.class != newvalue["type"].class
+        valid = false
+        errors << "Property `#{key}` must be of type `#{newvalue['type']}`(provided value `#{oldvalue}` of type `boolean`)"
+      end
+    end
```
Comment (orhantoy) [inline lib/ucfg.rb]: I'm not sure I understand this part entirely but I suspect there being a few issue with it. But let's handle that in the next round of reviews.

**Review by orhantoy — CHANGES_REQUESTED** [review]
Great work! 👏 

The type checking part seems to be sort of incorrect but let's address that in the next round of reviews. Firstly we should look into making the tests pass, there seems to be a small issue with whitespace.

Also, you'll notice that I made some changes in the PR, but they're mainly just concerned with Rubocop linting. Another time we should run `rubocop lib spec` to only check the files in lib and spec, the Gemfile, Rakefile, etc. should not be checked. I also updated the PR to target the `main` branch meaning when we merge this PR it will go into `main`.

File: lib/ucfg.rb (hunk 1 above)
Comment (SwathiSankararaman) [reply lib/ucfg.rb]: Removed line 12

File: lib/ucfg.rb (hunk 2 above)
Comment (SwathiSankararaman) [reply lib/ucfg.rb:47]: Added the empty line

File: lib/ucfg.rb (hunk 3 above)
Comment (SwathiSankararaman) [reply lib/ucfg.rb]: Implemented the correct logic

**Review by orhantoy — APPROVED** [review]

**Pull request merged**